### PR TITLE
Replace `@overload_glue` by `@overload` for `np.linspace` and `np.take`

### PR DESCRIPTION
**Pull request by guilhermeleobas**
ref: #8254 

Change `np.take` and `np.linspace` to use `@overload`

### CODE REVIEW

File: numba/np/arrayobj.py [hunk 1]
```diff
@@ -4545,21 +4531,28 @@ def impl(start, stop=None, step=None, dtype=None):
...
 
 
-@glue_lowering(np.linspace, types.Number, types.Number)
-def numpy_linspace_2(context, builder, sig, args):
+@overload(np.linspace)
+def numpy_linspace_2(start, stop):
+    if isinstance(start, types.Number) and isinstance(stop, types.Number):
+        def impl(start, stop):
+            return np.linspace(start, stop, 50)
+        return impl
```
Comment (stuartarchibald) [inline numba/np/arrayobj.py:4539]: I don't think that this version should exist. The default value for `num` should be set as a kwarg in the 3 arg variant, like `start, stop, num=50`.

File: numba/np/arrayobj.py [hunk 2]
```diff
@@ -4545,21 +4531,28 @@ def impl(start, stop=None, step=None, dtype=None):
...
+        return
 
-@glue_lowering(np.linspace, types.Number, types.Number, types.Integer)
-def numpy_linspace_3(context, builder, sig, args):
-    dtype = as_dtype(sig.return_type.dtype)
+    if any(isinstance(arg, types.Complex) for arg in [start, stop]):
+        dtype = types.complex128
+    else:
+        dtype = types.float64
+    dtype = as_dtype(dtype)
```
Comment (stuartarchibald) [inline numba/np/arrayobj.py:4554]:
```suggestion
    if any(isinstance(arg, types.Complex) for arg in [start, stop]):
        dtype = np.complex128
    else:
        dtype = np.float64
```
Just set the dtype directly?

File: numba/np/arrayobj.py [hunk 3]
```diff
@@ -4545,21 +4531,28 @@ def impl(start, stop=None, step=None, dtype=None):
...
 
-    # Implementation based on https://github.com/numpy/numpy/blob/v1.20.0/numpy/core/function_base.py#L24 # noqa: E501
```
Comment (stuartarchibald) [inline numba/np/arrayobj.py:4562]: This reference should be maintained?

Comment (guilhermeleobas) [reply numba/np/arrayobj.py:4562]: Done!

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch!

**Pull request merged**